graph/export.go

```diff
@@ -82,7 +83,9 @@ func (s *TagStore) CmdImageExport(job *engine.Job) error {
 	}
 	// write repositories, if there is something to write
 	if len(rootRepoMap) > 0 {
-		rootRepoJson, _ := json.Marshal(rootRepoMap)
+		tempBuffer := new(bytes.Buffer)
```

Comment:
@ankushagarwal the whole point of removing json.Marshal is to not use buffers :)

What you need to provide instead of tempBuffer, is a filedescriptor of the file we're writing to, which would remove the use of `ioutil.WriteFile`. You'd have to open with `os.Create` and use that result in lieu of tempBuffer.